api/admin/controller.py

```diff
@@ -610,6 +614,135 @@ def genres(self):
             })
         return data
 
+class DashboardController(CirculationManagerController):
+
+    def stats(self):
+        patron_count = self._db.query(Patron).count()
+
+        active_loans_patron_count = self._db.query(
+            distinct(Patron.id)
+        ).join(
+            Patron.loans
+        ).filter(
+            Loan.end >= datetime.now(),
+        ).count()
+
+        active_patrons = select(
+            [Patron.id]
+        ).select_from(
+            join(
+                Loan,
+                Patron,
+                and_(
+                    Patron.id == Loan.patron_id,
+                    Loan.id != None,
+                    Loan.end >= datetime.now()
+                )
+            )
+        ).union(
+            select(
+                [Patron.id]
+            ).select_from(
+                join(
+                    Hold,
+                    Patron,
+                    Patron.id == Hold.patron_id
+                )
+            )
+        ).alias()
+        
+
+        active_loans_or_holds_patron_count_query = select(
+            [func.count(distinct(active_patrons.c.id))]
+        ).select_from(
+            active_patrons
+        )
+
+        result = self._db.execute(active_loans_or_holds_patron_count_query)
+        active_loans_or_holds_patron_count = [r[0] for r in result][0]
+
+        loan_count = self._db.query(
+            Loan
+        ).filter(
+            Loan.end >= datetime.now()
+        ).count()
+
+        hold_count = self._db.query(Hold).count()
+
+        overdrive_count = self._db.query(
```

Comment:
It would be more compact and more extensible to iterate over a list of (DataSource, key name) pairs and build the dictionary that way. You're doing the same query for each data source and I think you want to omit a value if there are no active loans, since the reason is most likely the library has no books from that data source.